pkg/clients/elasticsearch.go

```diff
@@ -0,0 +1,199 @@
+package clients
+
+import (
+	"bytes"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/bitly/go-simplejson"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	contentTypeJSON = "application/json"
+)
+
+type MGetRequest struct {
+	Docs []MGetItem `json:"docs"`
+}
+
+type MGetItem struct {
+	Type string `json:"_type,omitempty"`
+	Id   string `json:"_id,omitempty"`
+}
+
+type MGetResponse struct {
+	Docs []MGetResponseItem `json:"docs"`
+}
+type MGetResponseItem struct {
+	Index   string                 `json:"_index,omitempty"`
+	Version int                    `json:"_version,omitempty"`
+	Found   bool                   `json:"found,omitempty"`
+	Source  map[string]interface{} `json:"_source,omitempty"`
+	MGetItem
+}
+
+//ElasticsearchClient is an admin client to query a local instance of Elasticsearch
+type ElasticsearchClient interface {
+	Get(path string) (string, error)
+	MGet(index string, items MGetRequest) (*MGetResponse, error)
+	Delete(path string) (string, error)
+	Put(path string, body string) (string, error)
+	Post(path string, body string) (string, error)
+}
+
+//DefaultElasticsearchClient is an admin client to query a local instance of Elasticsearch
+type DefaultElasticsearchClient struct {
+	serverURL string
+	client    *http.Client
+}
+
+//NewElasticsearchClient is the initializer to create an instance of ES client
+func NewElasticsearchClient(skipVerify bool, serverURL, adminCert, adminKey string, adminCA []string) (ElasticsearchClient, error) {
+	caCertPool := x509.NewCertPool()
```

Comment:
we should probably start with the `SystemCertPool()` instead so we can be sure to load any pod trusted CAs.